src/main/java/javax/validation/ElementKind.java

```diff
@@ -52,5 +53,12 @@
 	/**
 	 * The return value of a method or constructor.
 	 */
-	RETURN_VALUE
+	RETURN_VALUE,
+
+	/**
+	 * A type argument of a generic declaration.
+	 *
+	 * @since 2.0
+	 */
+	TYPE_ARGUMENT
```

Comment:
Taking a step back, how about "CONTAINER_ELEMENT"?

I like it because it's a bit more "semantical" and abstracts from the type argument/parameter/use debate. Also should be easier to grasp for people not familiar with the type system naming conventions.

It works well for constraints on parameterized types, array elements (which is the sore point for "type argument") and constraints given on wrappers (should the extractor decide to append a node).

"container element" focuses more on the "thing" that is constrained instead of the technical vehicle carrying the constraint annotation.